tools/toolchain/image

```diff
@@ -1 +1 @@
-docker.io/scylladb/scylla-toolchain:fedora-40-20240710
+docker.io/scylladb/scylla-toolchain:fedora-40-20240822
```

Comment:
Does this updated toolchain contain the optimized clang?

Reply:
It was based on the latest toolchain, should i do anything else?

Reply:
Include the prebuilt optimized clang

Reply:
how can i verify it? should i do it for every update of the toolchain?

Reply:
So it seems it's not with optimized clang, i will re-build it

Reply:
@avikivity Do we have a document explaining how to build a frozen toolchain with optimized clang?

Reply:
https://github.com/scylladb/scylla-enterprise/blob/b3f1b863cfa242a88a1e22ba315be5572ed692d7/tools/toolchain/prepare#L46

Reply:
yes this I have already done, is there another step? how can I verify it's working?

Reply:
It takes hours...

Reply:
Well, I meant how can I verify it inside the container? :-)

Reply:
clang_build/optimized_clang_${LLVM_CLANG_TAG}_${IMAGE_ID}.$(arch).tar.gz ?

Reply:
Yes, i have it as well.
```
[yaronkaikov@london]~/git/scylladb (update-node_exporter-1.8.2)$ ls -la clang_build/
total 988444
drwxr-xr-x  3 yaronkaikov yaronkaikov         89 Aug 23 00:53 .
drwxr-xr-x 59 yaronkaikov yaronkaikov       8192 Aug 23 00:02 ..
drwxr-xr-x 29 yaronkaikov yaronkaikov       4096 Aug 23 00:53 llvm-project
-rw-r--r--  1 yaronkaikov yaronkaikov 1012149341 Aug 23 00:54 optimized_clang_18.1.6_fedora-40-20240822.x86_64.tar.gz
```
Does it mean that the new container is good? or should i do anything else?

Reply:
Looks like it's there.

Reply:
There's a prebuilt one, see c93e2662ae11f9d69db055e5515414c54e419415

Reply:
@avikivity I downloaded the prebuilt as you mentioned above and then tried to build the toolchain with the following command `./tools/toolchain/prepare --clang-build-mode INSTALL_FROM --clang-archive ./clang_build/clang-18.1.6-Fedora-40-x86_64.tar.gz`

Reply:
@avikivity I downloaded the prebuilt as you mentioned above and then tried to build the toolchain with the following command `./tools/toolchain/prepare --clang-build-mode INSTALL_FROM --clang-archive ./clang_build/clang-18.1.6-Fedora-40-x86_64.tar.gz`

I got the following error: 
```
STEP 8/9: RUN ./tools/toolchain/optimized_clang.sh
+ case "${CLANG_BUILD}" in
CLANG_BUILD: INSTALL_FROM
+ echo 'CLANG_BUILD: INSTALL_FROM'
+ [[ -z clang_build/clang-18.1.6-Fedora-40-x86_64.tar.gz ]]
++ arch
+ ARCH=x86_64
+ [[ x86_64 = \x\8\6\_\6\4 ]]
+ LLVM_TARGET_ARCH=X86
+ SCYLLA_DIR=/mnt
+ CLANG_ROOT_DIR=/mnt/clang_build
+ CLANG_CHECKOUT_NAME=llvm-project
+ CLANG_BUILD_DIR=/mnt/clang_build/llvm-project
+ SCYLLA_BUILD_DIR=build_profile
+ SCYLLA_NINJA_FILE=build_profile.ninja
+ SCYLLA_BUILD_DIR_FULLPATH=/mnt/build_profile
+ SCYLLA_NINJA_FILE_FULLPATH=/mnt/build_profile.ninja
+ LLVM_CLANG_TAG=18.1.6
+ CLANG_SUFFIX=18
++ cd /mnt
++ realpath -m clang_build/clang-18.1.6-Fedora-40-x86_64.tar.gz
+ CLANG_ARCHIVE=/mnt/clang_build/clang-18.1.6-Fedora-40-x86_64.tar.gz
+ CLANG_OPTS=(-G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER="/usr/bin/clang" -DCMAKE_CXX_COMPILER="/usr/bin/clang++" -DLLVM_USE_LINKER="/usr/bin/ld.lld" -DLLVM_TARGETS_TO_BUILD="${LLVM_TARGET_ARCH};WebAssembly" -DLLVM_TARGET_ARCH="${LLVM_TARGET_ARCH}" -DLLVM_INCLUDE_BENCHMARKS=OFF -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF -DLLVM_ENABLE_BINDINGS=OFF -DLLVM_ENABLE_PROJECTS="clang" -DLLVM_ENABLE_RUNTIMES="compiler-rt" -DLLVM_ENABLE_LTO=Thin -DCLANG_DEFAULT_PIE_ON_LINUX=OFF -DLLVM_BUILD_TOOLS=OFF -DLLVM_VP_COUNTERS_PER_SITE=6 -DLLVM_BUILD_LLVM_DYLIB=ON -DLLVM_LINK_LLVM_DYLIB=ON -DCMAKE_INSTALL_PREFIX="/usr/local" -DLLVM_LIBDIR_SUFFIX=64 -DLLVM_INSTALL_TOOLCHAIN_ONLY=ON)
+ SCYLLA_OPTS=(--date-stamp "$(date "+%Y%m%d")" --debuginfo 1 --tests-debuginfo 1 --c-compiler="${CLANG_BUILD_DIR}/build/bin/clang" --compiler="${CLANG_BUILD_DIR}/build/bin/clang++" --build-dir="${SCYLLA_BUILD_DIR}" --out="${SCYLLA_NINJA_FILE}")
++ date +%Y%m%d
+ [[ INSTALL_FROM = \I\N\S\T\A\L\L ]]
+ [[ INSTALL_FROM = \I\N\S\T\A\L\L\_\F\R\O\M ]]
+ mkdir -p /mnt/clang_build
+ tar -C /mnt/clang_build -xpzf /mnt/clang_build/clang-18.1.6-Fedora-40-x86_64.tar.gz
+ cd /mnt/clang_build/llvm-project
+ ninja -C build install-distribution-stripped
ninja: Entering directory `build'
[1/3136] Building CXX object lib/Demangle/CMakeFiles/LLVMDemangle.dir/Demangle.cpp.o
FAILED: lib/Demangle/CMakeFiles/LLVMDemangle.dir/Demangle.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Demangle -I/mnt/clang_build/llvm-project/llvm/lib/Demangle -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Demangle/CMakeFiles/LLVMDemangle.dir/Demangle.cpp.o -MF lib/Demangle/CMakeFiles/LLVMDemangle.dir/Demangle.cpp.o.d -o lib/Demangle/CMakeFiles/LLVMDemangle.dir/Demangle.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Demangle/Demangle.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[2/3136] Building CXX object lib/Demangle/CMakeFiles/LLVMDemangle.dir/ItaniumDemangle.cpp.o
FAILED: lib/Demangle/CMakeFiles/LLVMDemangle.dir/ItaniumDemangle.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Demangle -I/mnt/clang_build/llvm-project/llvm/lib/Demangle -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Demangle/CMakeFiles/LLVMDemangle.dir/ItaniumDemangle.cpp.o -MF lib/Demangle/CMakeFiles/LLVMDemangle.dir/ItaniumDemangle.cpp.o.d -o lib/Demangle/CMakeFiles/LLVMDemangle.dir/ItaniumDemangle.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Demangle/ItaniumDemangle.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[3/3136] Building CXX object lib/Demangle/CMakeFiles/LLVMDemangle.dir/DLangDemangle.cpp.o
FAILED: lib/Demangle/CMakeFiles/LLVMDemangle.dir/DLangDemangle.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Demangle -I/mnt/clang_build/llvm-project/llvm/lib/Demangle -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Demangle/CMakeFiles/LLVMDemangle.dir/DLangDemangle.cpp.o -MF lib/Demangle/CMakeFiles/LLVMDemangle.dir/DLangDemangle.cpp.o.d -o lib/Demangle/CMakeFiles/LLVMDemangle.dir/DLangDemangle.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Demangle/DLangDemangle.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[4/3136] Building CXX object lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangle.cpp.o
FAILED: lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangle.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Demangle -I/mnt/clang_build/llvm-project/llvm/lib/Demangle -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangle.cpp.o -MF lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangle.cpp.o.d -o lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangle.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Demangle/MicrosoftDemangle.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[5/3136] Building CXX object lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangleNodes.cpp.o
FAILED: lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangleNodes.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Demangle -I/mnt/clang_build/llvm-project/llvm/lib/Demangle -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangleNodes.cpp.o -MF lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangleNodes.cpp.o.d -o lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangleNodes.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Demangle/MicrosoftDemangleNodes.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[6/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/ARMAttributeParser.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/ARMAttributeParser.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/ARMAttributeParser.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/ARMAttributeParser.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/ARMAttributeParser.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/ARMAttributeParser.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[7/3136] Building CXX object lib/Demangle/CMakeFiles/LLVMDemangle.dir/RustDemangle.cpp.o
FAILED: lib/Demangle/CMakeFiles/LLVMDemangle.dir/RustDemangle.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Demangle -I/mnt/clang_build/llvm-project/llvm/lib/Demangle -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Demangle/CMakeFiles/LLVMDemangle.dir/RustDemangle.cpp.o -MF lib/Demangle/CMakeFiles/LLVMDemangle.dir/RustDemangle.cpp.o.d -o lib/Demangle/CMakeFiles/LLVMDemangle.dir/RustDemangle.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Demangle/RustDemangle.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[8/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Allocator.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/Allocator.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/Allocator.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/Allocator.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/Allocator.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/Allocator.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[9/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/APInt.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/APInt.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/APInt.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/APInt.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/APInt.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/APInt.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[10/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/APFloat.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/APFloat.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/APFloat.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/APFloat.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/APFloat.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/APFloat.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[11/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/AMDGPUMetadata.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[12/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/APFixedPoint.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/APFixedPoint.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/APFixedPoint.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/APFixedPoint.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/APFixedPoint.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/APFixedPoint.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[13/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/APSInt.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/APSInt.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/APSInt.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/APSInt.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/APSInt.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/APSInt.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[14/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/ARMBuildAttrs.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/ARMBuildAttrs.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/ARMBuildAttrs.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/ARMBuildAttrs.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/ARMBuildAttrs.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/ARMBuildAttrs.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[15/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Base64.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/Base64.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/Base64.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/Base64.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/Base64.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/Base64.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[16/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/ARMWinEH.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/ARMWinEH.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/ARMWinEH.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/ARMWinEH.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/ARMWinEH.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/ARMWinEH.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[17/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/BalancedPartitioning.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/BalancedPartitioning.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/BalancedPartitioning.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/BalancedPartitioning.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/BalancedPartitioning.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/BalancedPartitioning.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
[18/3136] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/BinaryStreamError.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/BinaryStreamError.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/mnt/clang_build/llvm-project/build/lib/Support -I/mnt/clang_build/llvm-project/llvm/lib/Support -I/mnt/clang_build/llvm-project/build/include -I/mnt/clang_build/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fprofile-instr-use="/mnt/clang_build/llvm-project/combined.prof" -flto=thin -Werror=global-constructors -O3 -DNDEBUG -std=c++17  -fno-exceptions -funwind-tables -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/BinaryStreamError.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/BinaryStreamError.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/BinaryStreamError.cpp.o -c /mnt/clang_build/llvm-project/llvm/lib/Support/BinaryStreamError.cpp
error: Error in reading profile /mnt/clang_build/llvm-project/combined.prof: No such file or directory
ninja: build stopped: subcommand failed.
Error: building at STEP "RUN ./tools/toolchain/optimized_clang.sh": while running runtime: exit status 1
```

Any idea why?

Reply:
Why are you running optimized_clang.sh in the first place?

Reply:
@syuu1228 please help

Reply:
> Why are you running optimized_clang.sh in the first place?

I am just using `./tools/toolchain/prepare --clang-build-mode INSTALL_FROM --clang-archive ./clang_build/clang-18.1.6-Fedora-40-x86_64.tar.gz`

Reply:
@syuu1228 Any idea why it's failing?